docs/content/ingestion/merge-segments.md

```diff
@@ -0,0 +1,93 @@
+
+|Field|Type|Description|Default|Required|
+|-----|----|-----------|-------|--------|
+|keepGap|Boolean.|Indicate whether Druid should merge segments that are non-contiguous.|false|no|
+|hadoopDependencyCoordinates|Array of String.|A list of Hadoop dependency coordinates that Druid will use, this property will override the default Hadoop coordinates. Once specified, Druid will look for those Hadoop dependencies from the location specified by `druid.extensions.hadoopDependenciesDir`|null|no|
+|tuningConfig|JSON Object.|This is exactly same as the tuningConfig specified in Hadoop Index Task. See [TuningConfig](../ingestion/batch-ingestion.html#tuningconfig).|null|no|
+|hadoopMergeSpecs|Array of HadoopMergeSpec.|A list of HadoopMergeSpec. Each data source will have its own HadoopMergeSpec. See below.|null|no|
+
+For each data source on which you want to enable automatic merge, you would need to provide a `HadoopMergeSpec` so that Druid knows how to reindex those segments.
+Here is what goes inside `HadoopMergeSpec`
+
+|Field|Type|Description|Default|Required|
+|-----|----|-----------|-------|--------|
+|dataSource|String.|The data source of segments that will be automatically merged.|Value has to be provided by user.|yes|
+|queryGranularity|String.|Defines the granularity of the query while loading data. See [Granularities](../querying/granularities.html).|NONE|no|
+|dimensions|Array of String.|Name of dimension columns to load.|The super set of dimensions in the existing segments.|no|
+|metricsSpec|Array of aggregators.|A list of [aggregators](../querying/aggregations.html).|Value has to be provided by user.|yes|
+
+Example:
+
+```json
+{
+  ... other Coordinator dynamic configs...
+  "mergeBytesLimit": 500000000,
+  "hadoopMergeConfig": {
+    "keepGap": true,
+    "hadoopDependencyCoordinates": null,
+    "tuningConfig": null,
+    "hadoopMergeSpecs": [
+      {
+        "dataSource": "wikipedia",
+        "queryGranularity": "DAY",
+        "dimensions": ["language"],
+        "metricsSpec": [
+          {
+            "type": "count",
+            "name": "count"
+          },
+          {
+            "type": "doubleSum",
+            "name": "added",
+            "fieldName": "added"
+          },
+          {
+            "type": "doubleSum",
+            "name": "deleted",
+            "fieldName": "deleted"
+          },
+          {
+            "type": "doubleSum",
+            "name": "delta",
+            "fieldName": "delta"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+With this configuration posted to [Coordinator Dynamic Configuration end point]("../configuration/coordinator.html#dynamic-configuration"), 
+Coordinator will automatically merge imbalanced segments whose data sources are "wikipedia". Once it finds enough small segments whose total
+size is greater equal than `mergeBytesLimit`(500MB in the example), it will submit a Hadoop Index Task to reindex intervals covered by those imbalanced segments, 
```

Comment:
Since we have different compression algorithms, it would be good to specify exactly what size this is talking about. I'm guessing it is the on-disk size of the segment itself.